src/checkers/flymake-collection-statix.el

```diff
@@ -0,0 +1,134 @@
+;;; flymake-collection-statix.el --- Statix diagnostic function -*- lexical-binding: t -*-
+
+;; Copyright (c) 2022 Valeriy Litkovskyy
+
+;; Permission is hereby granted, free of charge, to any person obtaining a copy
+;; of this software and associated documentation files (the "Software"), to deal
+;; in the Software without restriction, including without limitation the rights
+;; to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+;; copies of the Software, and to permit persons to whom the Software is
+;; furnished to do so, subject to the following conditions:
+
+;; The above copyright notice and this permission notice shall be included in all
+;; copies or substantial portions of the Software.
+
+;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+;; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+;; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+;; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+;; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+;; SOFTWARE.
+
+;;; Commentary:
+
+;; This file provides a flymake backend that lets the user analyze nix code
+;; using statix.
+
+;;;; Tips
+
+;; + You can customize `flymake-collection-statix-command' to your liking.
+
+;; + Some warnings have "{fix}" suffix.  They can be fixed by calling
+;;   `flymake-collection-statix-fix'.
+
+;;;; Credits
+
+;; This diagnostic would not have been possible without the statix[1] program.
+
+;;  [1] https://github.com/nerdypepper/statix
+
+;;; Code:
+
+;;;; Requirements
+
+(require 'flymake)
+(require 'flymake-collection)
+
+(eval-when-compile
+  (require 'flymake-collection-define))
+
+;;;; Customization
+
+(defcustom flymake-collection-statix-command
```

Comment:
I think something a little less customizable would be better. These arguments are the contract this checker needs to be able to parse the output correctly. Perhaps a `flymake-collection-statix-args` option which defaults to nil would be better. Any extra arguments users choose to put could go in there.

Reply:
Well, I made it a command because in the **Nix** world it is not uncommon to see commands that are not installed globally, but are something like:
`nix run nixpkgs#statix -- -other -args`.

I wouldn't object if you insist on consistency with the rest of the package, thou.

Reply:
Just adding my thoughts. When using as an editor integration, I feel like only very few people will be using it that way. They will either have it installed globally or within the local env with shell.nix/flakes. But if we are going this route, we can probably also consider a more generic approach to supporting things like `npx` for node based checkers.